scripts/run-e2e-tests.sh

```diff
@@ -75,18 +75,35 @@ function install_cluster() {
 
   # install kubectl
   echo "step 3/6: installing kubectl"
-  stable_release=$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)
-  exit_on_error "unable to retrieve latest stable version of kubectl"
-  curl -LO https://storage.googleapis.com/kubernetes-release/release/${stable_release}/bin/linux/amd64/kubectl \
-    && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
-  exit_on_error "install kubectl failed"
+
+  if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    echo "Installing Kubectl for Linux.."
+          stable_release=$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)
+          exit_on_error "unable to retrieve latest stable version of kubectl"
+          curl -LO https://storage.googleapis.com/kubernetes-release/release/${stable_release}/bin/linux/amd64/kubectl \
+          && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+          exit_on_error "install kubectl failed"
+  elif [[ "$OSTYPE" == "darwin"* ]]; then
+    echo "Installing Kubectl for Mac.."
+              curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/darwin/amd64/kubectl" \
+              && chmod +x kubectl && sudo mv kubectl /usr/local/bin
```

Comment:
We need to pull the version from the same location. We should not mix `dl.k8s.io` and `googleapis.com`
The full procedure is the same for both installs except for a small part of the URL. Instead of having two copied procedures we should have one and just replace `darwin` and `linux` when building the URL.